### PR TITLE
Find Nearby Station: Docks and Road

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Mostly a copy paste job from track (Good to see CS also copy pasted it)

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
To what extent can we template these station find functions? Or would you prefer to merge as-is now and refactor later?

**Comment by duncanspumpkin** [comment]
> To what extent can we template these station find functions? Or would you prefer to merge as-is now and refactor later?

Hmm not sure. The track and road ones are very similar but are different the ai/non-ai ones could probably be combined but at this point i feel it would just make things messy.

**Comment by duncanspumpkin** [comment]
> Code looks fine. Has this been tested for divergences in the replays yet?

good thing you asked. Seems there is a divergence.

File: src/OpenLoco/src/GameCommands/Road/CreateRoadStation.cpp [hunk 1]
```diff
@@ -23,26 +23,41 @@ namespace OpenLoco::GameCommands
...
 
-    struct NearbyStation
-    {
-        StationId id;
-        bool isPhysicallyAttached;
-    };
-
     // 0x004900B8
-    static NearbyStation sub_4900B8(World::Pos3 pos, uint16_t tad, uint8_t roadObjectId)
+    static StationManager::NearbyStation findNearbyStationOnRoad(World::Pos3 pos, uint16_t tad, uint8_t roadObjectId)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/Road/CreateRoadStation.cpp:27]: something wrong with this function

**Comment by duncanspumpkin** [comment]
Divergence due to an original bug in vehicle update updateRoadMotion. Will do that function first.

**Comment by duncanspumpkin** [comment]
With #2588 now done this no longer has any divergences (or any need to have a hack to prevent them)

**Pull request merged**